commit 1c8fb3ea40e681de04bc51c4bee101700ffbf4b0
Author: Lukas F. Hartmann <lukas@mntre.com>
Date:   Mon Sep 26 19:26:17 2022 +0200

    mnt-rkx7: video: add reform display video mode

diff --git a/litex/soc/cores/video.py b/litex/soc/cores/video.py
index f15ace23..f8a74fef 100644
--- a/litex/soc/cores/video.py
+++ b/litex/soc/cores/video.py
@@ -123,6 +123,17 @@ video_timings = {
         "v_sync_offset" : 4,
         "v_sync_width"  : 5,
     },
+    "1920x1080@rkx7": {
+        "pix_clk"       : 162e6,
+        "h_active"      : 1920,
+        "h_blanking"    : 159, # off by one in vtg
+        "h_sync_offset" : 40,
+        "h_sync_width"  : 40,
+        "v_active"      : 1080,
+        "v_blanking"    : 32,
+        "v_sync_offset" : 4,
+        "v_sync_width"  : 4,
+    },
     "1920x1200@60Hz": {
         "pix_clk"       : 148.2e6,
         "h_active"      : 1920,
